bot/exts/info/tags.py

```diff
@@ -24,145 +29,227 @@
 FOOTER_TEXT = f"To show a tag, type {constants.Bot.prefix}tags <tagname>."
 
 
+class COOLDOWN(enum.Enum):
+    """Sentinel value to signal that a tag is on cooldown."""
+
+    obj = object()
+
+
+class TagIdentifier(NamedTuple):
+    """Stores the group and name used as an identifier for a tag."""
+
+    group: Optional[str]
+    name: str
+
+    def get_fuzzy_score(self, fuzz_tag_identifier: TagIdentifier) -> float:
+        """Get fuzzy score, using `fuzz_tag_identifier` as the identifier to fuzzy match with."""
+        if self.group is None:
+            if fuzz_tag_identifier.group is None:
+                # We're only fuzzy matching the name
+                group_score = 1
+            else:
+                # Ignore tags without groups if the identifier contains a group
+                return .0
```

Comment:
Something I am a bit on the fence about, do we prefer `0.0` here?

Reply:
If I was writing it myself i'd probably write `0.0`, but both ways are valid and I don't think our style guide indicates any preferred way, so probably doesn't need changing.